x/ecocredit/client/testsuite/grpc.go

```diff
@@ -434,106 +350,49 @@ func (s *IntegrationTestSuite) TestGetSellOrders() {
-	val := s.network.Validators[0]
-	batchDenom := s.batchInfo.BatchDenom
-
-	testCases := []struct {
-		name     string
-		url      string
-		expErr   bool
-		errMsg   string
-		expItems int
-	}{
-		{
-			"invalid denom",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/batch-denom/%s", val.APIAddress, "foo"),
-			true,
-			"invalid denom",
-			0,
-		},
-		{
-			"valid request",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/batch-denom/%s", val.APIAddress, batchDenom),
-			false,
-			"",
-			3,
-		},
-		{
-			"valid request pagination",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/batch-denom/%s?pagination.limit=2", val.APIAddress, batchDenom),
-			false,
-			"",
-			2,
-		},
-	}
-
-	require := s.Require()
-	for _, tc := range testCases {
-		tc := tc
-		s.Run(tc.name, func() {
-			resp, err := rest.GetRequest(tc.url)
+			var res marketplace.QuerySellOrdersResponse
+			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &res)
 			require.NoError(err)
-
-			var sellOrders ecocredit.QuerySellOrdersByBatchDenomResponse
-			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &sellOrders)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
+			require.NotNil(res.SellOrders)
+			if tc.expItems > 0 {
+				require.Len(res.SellOrders, tc.expItems)
 			} else {
-				require.NoError(err)
-				require.NotNil(sellOrders.SellOrders)
-				require.Len(sellOrders.SellOrders, tc.expItems)
+				require.Greater(len(res.SellOrders), 0)
 			}
+
 		})
 	}
 }
 
-func (s *IntegrationTestSuite) TestGetSellOrdersByAddress() {
+func (s *IntegrationTestSuite) TestGetSellOrdersByBatchDenom() {
 	val := s.network.Validators[0]
+	_, _, batchDenom := s.createClassProjectBatch(val.ClientCtx, val.Address.String())
+	validAsk := types.NewInt64Coin(core.DefaultParams().AllowedAskDenoms[0].Denom, 10)
+	expiration, err := types2.ParseDate("expiration", "2090-10-10")
+	s.Require().NoError(err)
+	_, err = s.createSellOrder(val.ClientCtx, &marketplace.MsgSell{
+		Owner: val.Address.String(),
+		Orders: []*marketplace.MsgSell_Order{
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+		},
+	})
+	s.Require().NoError(err)
 
 	testCases := []struct {
 		name     string
 		url      string
-		expErr   bool
-		errMsg   string
 		expItems int
 	}{
-		{
-			"invalid address",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/address/%s", val.APIAddress, "abc"),
-			true,
-			"invalid request",
-			0,
-		},
 		{
 			"valid request",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/address/%s", val.APIAddress, val.Address.String()),
-			false,
-			"",
-			3,
+			fmt.Sprintf("%s%ssell-orders/batch-denom/%s", val.APIAddress, marketplaceRoute, batchDenom),
+			-1,
```

Comment:
We should not be expecting a negative number of items.

Reply:
see above